### DISCUSSION TITLE
Milvus - Issue while creating multiple vectors in the same collection

### DISCUSSION THREAD

**Discussion by tfiroz11** [milvus-io/milvus#30803] [Q&A and General discussion]
Hi Team, 

I have a scenario where I need to enable search capabilities based on the two fields. When I am trying to creating index I am getting below error..Could you please let me know the best way to handle this scenario?

**multiple vector fields is not supported, fields name: answerembedding, embedding**

Regards,
Tahseen

**Comment by xiaofan-luan** [comment]
Hybrid Search on multi embeddings is on our roadmap and will be released on 2.4.
You can track it here
https://github.com/milvus-io/milvus/issues/25639

**Reply by wrosko** [reply]
HI @xiaofan-luan thanks for your team's great work. A question on this: will this mean we will be able to store multiple vector fields in a collection (running milvus standalone/cluster on our dev/test to see if we want to go with Milvus or another vector db in the future).

Our use case would be: we have different embedding methods being run on a set of data and we would like to be able to store them within the same collection. However, going with this paradigm means that we'd have to tear them down and re-build the collection if we introduce a new embedding method and want to store it. We'd want this feature mostly for organization purposes.

ex: given a set of documents, let's call them "all_posts", we want to get the openai ada embeddings,  sentence-transformer embeddings, and maybe another type. storing them all in the "all_posts" collection could be nice. We would want to search on a single field at a time, not a multi embedding search

**Reply by xiaofan-luan** [reply]
Hi @wrosko 
so far your understanding is correct. you will need to recreate a colleciton with more fields. and rebuild the collection.

We plan to add dynamic schema change in Milvus 3.0. While you could add more field dynamically on existing collections. So far there is nothing you could do. Maybe create multi collections with same Primary key is a good idea.

**Comment by ssling0817** [comment]
Hello, I have updated the pymilvus to version 2.4.1 and tried to connect to the milvus server, but the error message is still there. Do the server also need to be updated? Or updating code is enough. Thanks!

`MilvusException: <MilvusException: (code=65535, message=multiple vector fields is not supported, fields name: summary_embeddings, description_embeddings)>`

**Reply by yhmo** [reply]
Both pymilvus and milvus server need to be upgraded to v2.4.x.
